x-pack/plugins/security_solution/server/endpoint/routes/trusted_apps/handlers.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RequestHandler } from 'kibana/server';
+import {
+  GetTrustedAppsListRequest,
+  GetTrustedListAppsResponse,
+} from '../../../../common/endpoint/types';
+import { EndpointAppContext } from '../../types';
+import { exceptionItemToTrustedAppItem } from './utils';
+
+export const getTrustedAppsListRouteHandler = (
+  endpointAppContext: EndpointAppContext
+): RequestHandler<undefined, GetTrustedAppsListRequest> => {
+  const exceptionsListService = endpointAppContext.service.getExceptionsList();
+  const logger = endpointAppContext.logFactory.get('trusted_apps');
+
+  return async (context, req, res) => {
+    const { page, per_page: perPage } = req.query;
+
+    // FIXME:PT call create list once PR merges
+
+    try {
+      const results = await exceptionsListService?.findExceptionListItem({
+        listId: 'trusted-apps', // FIXME:PT get value from const defined in pending PR
+        page,
+        perPage,
+        filter: '',
+        namespaceType: 'agnostic',
+        sortField: 'name',
+        sortOrder: 'asc',
+      });
+      const body: GetTrustedListAppsResponse = {
+        data: results?.data.map(exceptionItemToTrustedAppItem) ?? [],
+        total: results?.total ?? 0,
+        page: results?.page ?? 1,
+        per_page: perPage!,
+      };
+      return res.ok({ body });
+    } catch (error) {
+      logger.error(error);
+      return res.internalError({ body: error });
```

Comment:
I'm not sure do we follow REST conventions for response codes? I can imagine there are errors that are client errors that would trigger 400 or 403.

Reply:
I think if you come here, we don't know if its a `400` (I belive the platform may handle 403). These I think these should be `500`'s.

Reply:
if we can prove for sure it was a client error then it should definitely be 4xx error. And I believe a few of those cases are already caught (not authorized: 403, due to `authRequired: true`, though I'm not sure if we also need the `tags: ['access:securitySolution']`; and schema should catch bad parameters and give some `4xx` status codes, probably 400).

If we can't prove it's a request fault, better to say 'excuse me' (5xx) than "I can't believe you've done this" (4xx)

Reply:
We can revisit this later to see if it needs more interpretation. but given what is inside of the `try {}`, I'm not sure just how much logic we would need to decipher if the `error` that came in here in order to determine if a `4xx` is appropriate. This block here is a "catch all" - maybe you guys see a missing `try {}` inside of the above code that I might be missing where its `catch()` should return a `4xx`?